common/src/com/thoughtworks/go/util/SvnLogXmlParser.java

```diff
@@ -56,29 +56,36 @@
         for (Iterator iterator = logEntries.iterator(); iterator.hasNext();) {
             Element logEntry = (Element) iterator.next();
 
-            modifications.add(parseLogEntry(logEntry, path));
+            modification = parseLogEntry(logEntry, path);
+            if (modification != null) {
+                modifications.add(modification);
+            }
         }
 
         return modifications;
     }
 
     private Modification parseLogEntry(Element logEntry, String path) throws ParseException {
+        Element logEntryPaths = logEntry.getChild("paths");
+        if (logEntryPaths == null) {
+            /* Path-based access control forbids us from learning
+             * details of this log entry, so skip it. */
+            return null;
+        }
+
```

Comment:
Can we write a test for this at?
https://github.com/gocd/gocd/blob/81701742adecc398e11d2a04443eff3f7bfc9cff/common/test/unit/com/thoughtworks/go/util/SvnLogXmlParserTest.java